server/src/main/java/org/elasticsearch/index/mapper/vectors/NormalizedCosineFloatVectorValues.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper.vectors;
+
+import org.apache.lucene.index.FloatVectorValues;
+import org.apache.lucene.index.NumericDocValues;
+
+import java.io.IOException;
+
+import static org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper.isNotUnitVector;
+
+public class NormalizedCosineFloatVectorValues extends FloatVectorValues {
```

Comment:
Should this class be called `DenormalizeCosineFloatVectorValues` as we provide their denormalized values?

Reply:
good call @mayya-sharipova